collectionService.js

```diff
@@ -410,7 +414,32 @@ var CollectionService = module.exports = BaseService.extend( {
 			this.length++;
 		}
 
-		_.extend( this._recordsById[ recordId ], dto );
+		for( field in dto ) {
+			if( _.isObject( dto[ field ] ) ){
+				this._recordsById[ recordId ][ field ] = _.clone( dto[ field ] );
+				this._deepFreeze( this._recordsById[ recordId ][ field ] );
+			} else this._recordsById[ recordId ][ field ] = dto[ field ];			
+		}
+	},
+
+	// From : https://github.com/jsdf/deep-freeze
+	_deepFreeze( obj ) {
+		Object.freeze( obj );
+	
+		var objIsFunction = typeof obj === "function";
+		var hasOwnProp = Object.prototype.hasOwnProperty;
+	
+		Object.getOwnPropertyNames( obj ).forEach( function ( property ) {
+			if ( hasOwnProp.call( obj, property )
+			&& ( objIsFunction ? property !== 'caller' && property !== 'callee' && property !== 'arguments' : true )
+			&& obj[ prop ] !== null
+			&& ( typeof obj[ property ] === "object" || typeof obj[ property ] === "function" )
+			&& !Object.isFrozen( obj[ property ] ) ) {
```

Comment:
this is not necessary either

Reply:
Yeah, we should not have shared attributes, circular references or partially cloned objects. Gone

Reply:
thanks!